workflow/docker_agent_workflow-final.cwl

```diff
@@ -0,0 +1,271 @@
+      - id: synapse_config
+        source: "#synapseConfig"
+    out:
+      - id: uploaded_fileid
+      - id: uploaded_file_version
+      - id: results
+
+  annotate_docker_upload_results:
+    run: https://raw.githubusercontent.com/Sage-Bionetworks/ChallengeWorkflowTemplates/v2.2/annotate_submission.cwl
+    in:
+      - id: submissionid
+        source: "#submissionId"
+      - id: annotation_values
+        source: "#upload_results/results"
+      - id: to_public
+        default: true
+      - id: force_change_annotation_acl
+        default: true
+      - id: synapse_config
+        source: "#synapseConfig"
+      - id: previous_annotation_finished
+        source: "#annotate_docker_validation_with_output/finished"
+    out: [finished]
+
+  validation:
+    run: validate.cwl
+    in:
+      - id: inputfile
+        source: "#run_docker/predictions"
+      - id: goldstandard
+        source: "#download_goldstandard/filepath"
+    out:
+      - id: results
+      - id: status
+      - id: invalid_reasons
+  
+  validation_email:
+    run: https://raw.githubusercontent.com/Sage-Bionetworks/ChallengeWorkflowTemplates/v2.2/validate_email.cwl
+    in:
+      - id: submissionid
+        source: "#submissionId"
+      - id: synapse_config
+        source: "#synapseConfig"
+      - id: status
+        source: "#validation/status"
+      - id: invalid_reasons
+        source: "#validation/invalid_reasons"
+      - id: errors_only
+        default: true
+    out: [finished]
+
+  annotate_validation_with_output:
+    run: https://raw.githubusercontent.com/Sage-Bionetworks/ChallengeWorkflowTemplates/v2.2/annotate_submission.cwl
+    in:
+      - id: submissionid
+        source: "#submissionId"
+      - id: annotation_values
+        source: "#validation/results"
+      - id: to_public
+        default: true
+      - id: force_change_annotation_acl
+        default: true
+      - id: synapse_config
+        source: "#synapseConfig"
+      - id: previous_annotation_finished
+        source: "#annotate_docker_upload_results/finished"
+    out: [finished]
+
+  check_status:
+    run: https://raw.githubusercontent.com/Sage-Bionetworks/ChallengeWorkflowTemplates/v2.2/check_status.cwl
+    in:
+      - id: status
+        source: "#collect_validation/status"
+      - id: previous_annotation_finished
+        source: "#annotate_validation_with_output/finished"
+      - id: previous_email_finished
+        source: "#validation_email/finished"
+    out: [finished]
+
+  scoring:
+    run: score.cwl
+    in:
+      - id: inputfile
+        source: "#run_docker/predictions"
+      - id: goldstandard
+        source: "#download_goldstandard/filepath"
+      - id: check_validation_finished
+        source: "#check_status/finished"
+      - id: dataset
+        source: "#dataset"
+    out:
+      - id: results
+
+  score_email:
+    run: https://raw.githubusercontent.com/Sage-Bionetworks/ChallengeWorkflowTemplates/v2.2/score_email.cwl
```

Comment:
Are we emailing scores back?

Reply:
oh i was keeping it for now while i do the dry runs. but no, we won't be.